tests/integration/src/test/resources/org/mule/test/integration/client/client-dispatch-javamessageprocessor-flow.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:vm="http://www.mulesoft.org/schema/mule/vm"
+      xmlns:spring="http://www.springframework.org/schema/beans"
+      xmlns:http="http://www.mulesoft.org/schema/mule/http"
+      xsi:schemaLocation="
+           http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+           http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-current.xsd
+           http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+           http://www.mulesoft.org/schema/mule/vm http://www.mulesoft.org/schema/mule/vm/current/mule-vm.xsd">
+
+    <vm:connector name="vmQueues"/>
+    <http:listener-config name="HTTP_Listener_Configuration" host="0.0.0.0" port="${port1}"/>
+
+    <flow name="innerFlow">
+        <vm:inbound-endpoint exchange-pattern="one-way" path="vminnertest"/>
+        <logger level="WARN"/>
+    </flow>
+
+    <spring:beans>
+        <spring:bean id="myProcessor"
+                     class="org.mule.test.integration.client.MuleClientDispatchFromJavaMessageProcessorTestCase$FlowVarsDispatchFlowUsingNewMuleClient"/>
+    </spring:beans>
+
+    <flow name="flowVarsFlowUsingVM">
+        <vm:inbound-endpoint exchange-pattern="request-response" path="vmflowvars"/>
+        <set-variable variableName="team" value="Sales"/>
+        <processor ref="myProcessor"/>
+        <set-variable variableName="ammount" value="100"/>
+        <set-payload value="#[flowVars.team + ': ' + flowVars.ammount]"/>
```

Comment:
Consider using FlowAsserts.verify("flowVarsFlowUsingVM") and <test:assert/> processors as it make the test much more readable/direct.